blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx

```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
 		rightRailBottom,
 		fullWidth2,
 		footer,
-	] = children;
+	] = React.Children.toArray(children);
 	const featureList = useFeatureList();
 
 	return (
-		<>
-			<header className="page-header">{navigation}</header>
-			<section role="main" id="main" className="main" tabIndex="-1">
-				<div className="container layout-section">
-					<div className="row">
-						<div className="col-sm-xl-12 layout-section wrap-bottom">{fullwidth1}</div>
-					</div>
-					<div className="row advanced-grid">
-						<section className="col-sm-md-12 col-lg-xl-8 advanced-grid-desktop-main-area">
-							<RenderChild Item={main} tabletPlacement="2" />
-							<RenderChild Item={main2} tabletPlacement="4" />
-						</section>
-						<section className="col-sm-md-12 col-lg-xl-4 layout-section">
-							<RenderChild Item={rightRailTop} tabletPlacement="1" />
-							<RenderChild Item={rightRailMiddle} tabletPlacement="3" />
-							<RenderChild Item={rightRailBottom} tabletPlacement="5" />
-						</section>
-					</div>
-					{featureList["7"] > 0 && (
-						<div className="row">
-							<div className="col-sm-xl-12 layout-section wrap-bottom">{fullWidth2}</div>
-						</div>
-					)}
-				</div>
+		<div className={LAYOUT_CLASS_NAME}>
+			{navigation ? (
+				<Stack role="banner" className={`${LAYOUT_CLASS_NAME}__navigation`}>
+					{navigation}
+				</Stack>
+			) : null}
+
+			<section role="main" tabIndex="-1" className={`${LAYOUT_CLASS_NAME}__main`}>
+				{fullwidth1 ? (
+					<Stack className={`${LAYOUT_CLASS_NAME}__full-width-1`}>{fullwidth1}</Stack>
+				) : null}
+
+				<Grid className={`${LAYOUT_CLASS_NAME}__rail-container`}>
+					<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item`}>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item-1`}>{main}</Stack>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item-2`}>{main2}</Stack>
+					</Stack>
+					<Stack role="complementary" className={`${LAYOUT_CLASS_NAME}__main-side-rail`}>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-top`}>{rightRailTop}</Stack>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-middle`}>
+							{rightRailMiddle}
+						</Stack>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-bottom`}>
+							{rightRailBottom}
+						</Stack>
+					</Stack>
+				</Grid>
+				{featureList["7"] > 0 ? (
```

Comment:
This line with the ternary is the only one preventing full test coverage.
Do we need a test for this?
```
> jest --changedSince=origin/arc-themes-release-version-2.00 --coverage --passWithNoTests

 PASS  blocks/right-rail-block/layouts/right-rail/default.test.jsx (9.772 s)
 PASS  blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.test.jsx (9.775 s)
-------------------------------------------------------|---------|----------|---------|---------|-------------------
File                                                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------------------------------------------|---------|----------|---------|---------|-------------------
All files                                              |     100 |    95.83 |     100 |     100 |
 right-rail-advanced-block/layouts/right-rail-advanced |     100 |    91.66 |     100 |     100 |
  default.jsx                                          |     100 |    91.66 |     100 |     100 | 61
 right-rail-block/layouts/right-rail                   |     100 |      100 |     100 |     100 |
  default.jsx                                          |     100 |      100 |     100 |     100 |
-------------------------------------------------------|---------|----------|---------|---------|-------------------
```

Reply:
updated [6857e5f](https://github.com/WPMedia/arc-themes-blocks/pull/1335/commits/6857e5f7cd3e167a4cd55fbb6e3d250b008729e3) good q!

Reply:
I know this isn't you, but what even is featureList["7"] ?

Reply:
nm, I figured it out... I do not like this code :p

Reply:
Cool, got the update. 100% coverage!

Reply:
@JackHowa it's not you, it's the system